src/sfdt/toggleBookmark.ts

```diff
@@ -5,6 +5,88 @@ import {
 	isBookmarkStart,
 	isBookmarkEnd,
 } from '../queryBookmark'
+import Stack from '../stack'
+
+const makeToggleOff = (inlines, name) => {
+  const newInlines = []; // Should act as queue for getting new list of inlines
+  const stackOfBookmarks = new Stack(); // will be used to find relationship of bookmarks
+  let toggleOffAtIndex = null;
+
+  inlines.forEach((inline, index) => {
+    const nextInline = inlines[index + 1];
+    const prevInline = inlines[index - 1];
+
+    if (!(inline.fieldType || inline.hasFieldEnd)) {
```

Comment:
sfdt query logic to determine what type of an inilne we are dealing with should live in a separate re-useable function, for an example please see the queryBookmark file.